### PR TITLE
Update header UI

### PR DESCRIPTION
**Pull request by daviddossett**
This PR addresses the following:
- Updates header styles and layout to match the design in  #4029 and aligns with extension detail page in core
- Uses extension detail- like buttons instead of mixed icons/large buttons
- Ensures content reflows on small viewport/editor sizes
- Updates max width of overview content
- Refactors header JSX by extracting out `Title`, `Subtitle`, and `ButtonGroup` into their own components.

## Before

CleanShot 2022-10-13 at 11 16 32@2x

## After

CleanShot 2022-10-13 at 11 05 35@2x

## Demo

https://user-images.githubusercontent.com/25163139/195673197-6470f0fc-3760-4f3d-a472-bf03f202d156.mp4

@alexr00 I didn't have the time but I think we should replace the title editing form with a quick pick like how Settings Profile rename works. Thoughts?

cc @digitarald

### CODE REVIEW

File: webviews/components/header.tsx [hunk 1]
```diff
@@ -66,50 +67,66 @@ function Title({ title, number, url, canEdit, isCurrentlyCheckedOut, isIssue, re
...
+function Subtitle({ state, isDraft, isIssue, author, base, head }) {
+	return (
+		<div className="subtitle">
+			<div id="status">{getStatus(state, isDraft)}</div>
+			<div className="author">
+				{!isIssue ? <Avatar for={author} /> : null}
+				{!isIssue ? (
+					<div className="merge-branches">
+						<AuthorLink for={author} /> {getActionText(state)} into{' '}
+						<code className="branch-tag"> {base}</code> from <code className="branch-tag"> {head} </code>
```
Comment (rzhao271) [inline webviews/components/header.tsx:127]: Not sure about the spacing here. Is there a difference when there's a space in the code tag and a space outside? And, does the space after `{head}` add padding?

**Review by rzhao271 — APPROVED** [review]

File: webviews/components/header.tsx [hunk 2]
```diff
@@ -66,50 +67,66 @@ function Title({ title, number, url, canEdit, isCurrentlyCheckedOut, isIssue, re
...
+	const editableTitle = inEditMode ? titleForm : displayTitle;
+	return editableTitle;
+}
+
+function ButtonGroup({ isCurrentlyCheckedOut, canEdit, isIssue, repositoryDefaultBranch, setEditMode }) {
+	const { refresh, copyPrLink } = useContext(PullRequestContext);
+
+	return (
+		<div className="button-group">
+			<CheckoutButtons {...{ isCurrentlyCheckedOut, isIssue, repositoryDefaultBranch }} />
```
Comment (rzhao271) [inline webviews/components/header.tsx:100]: I'm a bit confused on how the `{...{a, b, c}}` part works. I'd have expected something like `{...[a, b, c]}` considering how the variables were already extracted in the function parameter.

**Comment by Thomas1664** [comment]
A visual distinction between drafts and open PRs like on the GitHub website would be nice, i.e. by making the `open` label green

File: webviews/components/header.tsx [hunk 3]
```diff
@@ -66,50 +67,66 @@ function Title({ title, number, url, canEdit, isCurrentlyCheckedOut, isIssue, re
...
+				</a>
+			</h2>
+		</div>
+	);
+
+	const editableTitle = inEditMode ? titleForm : displayTitle;
+	return editableTitle;
+}
+
+function ButtonGroup({ isCurrentlyCheckedOut, canEdit, isIssue, repositoryDefaultBranch, setEditMode }) {
```
Comment (Thomas1664) [inline webviews/components/header.tsx:95]: Can you move this to its own file? I want to use it in #4057 as well.

Comment (daviddossett) [reply webviews/components/header.tsx:95]: Could you create an issue? This button group isn't really generic (yet) and I'm keeping these PRs as scoped as possible.

File: webviews/components/header.tsx (hunk 2 above)
Comment (daviddossett) [reply webviews/components/header.tsx:100]: This syntax was already used in those components so I'll leave this for a follow up PR.

File: webviews/components/header.tsx (hunk 1 above)
Comment (daviddossett) [reply webviews/components/header.tsx]: Seems like it had no effect on render but I cleaned it out anyway.

**Comment by Thomas1664** [comment]
@daviddossett In #4052 I informed you that your PR breaks icon button focus outline, but you didn't take my concern serious: https://github.com/microsoft/vscode-pull-request-github/pull/4052#discussion_r996013349

After merging your changes from main into #4057 I noticed that my previously added support for focus is no longer visible. I hope that you take my concern seriously now and fix the issue that was caused by your PR.

**Review by mjbvz — APPROVED** [review]

**Comment by daviddossett** [comment]
@Thomas1664 where are you still seeing the issue? It sounded like it was fixed by this PR removing the icons from the header.

**Comment by Thomas1664** [comment]
> @Thomas1664 where are you still seeing the issue? It sounded like it was fixed by this PR removing the icons from the header.

In the comment action bar. But you have to check out #4057 because the action buttons weren't visible before on focus.

**Comment by daviddossett** [comment]
I already fixed it in #4066

**Comment by alexr00** [comment]
> I didn't have the time but I think we should replace the title editing form with a quick pick like how Settings Profile rename works. Thoughts?

I have a slight preference for leaving it as is. When you edit a comment in the overview, you get the text area. Keeping the title the same way is consistent.

**Pull request merged**